sway 0.12-rc1 compile problems with GCC7

```
/builddir/build/BUILD/sway-0.12-rc1/swaylock/main.c: In function 'notify_key':
/builddir/build/BUILD/sway-0.12-rc1/swaylock/main.c:173:31: error: this statement may fall through [-Werror=implicit-fallthrough=]
     if (sym == XKB_KEY_Escape || xkb_state_mod_name_is_active(registry->input->xkb.state,
         ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       XKB_MOD_NAME_CTRL, XKB_STATE_MODS_EFFECTIVE) > 0) {
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/sway-0.12-rc1/swaylock/main.c:185:3: note: here
   default:
   ^~~~~~~
```
```
/builddir/build/BUILD/sway-0.12-rc1/sway/handlers.c: In function 'handle_key':
/builddir/build/BUILD/sway-0.12-rc1/sway/handlers.c:780:8: error: this statement may fall through [-Werror=implicit-fallthrough=]
     if (!binding->release && valid_bindsym(binding)) {
        ^
/builddir/build/BUILD/sway-0.12-rc1/sway/handlers.c:784:4: note: here
    case WLC_KEY_STATE_RELEASED:
    ^~~~
/builddir/build/BUILD/sway-0.12-rc1/sway/handlers.c: In function 'handle_view_created':
/builddir/build/BUILD/sway-0.12-rc1/sway/handlers.c:448:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
   newview = new_floating_view(handle);
   ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/sway-0.12-rc1/sway/handlers.c:449:2: note: here
  case WLC_BIT_POPUP:
  ^~~~
```

[Build environment creation log](https://kojipkgs.fedoraproject.org//work/tasks/6760/18056760/root.log)
[Compelte build log](https://kojipkgs.fedoraproject.org//work/tasks/6760/18056760/build.log)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,7 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wextra")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-parameter")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wimplicit-fallthrough=0")
 
 list(INSERT CMAKE_MODULE_PATH 0
 	${CMAKE_CURRENT_SOURCE_DIR}/CMake

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,11 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wextra")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-parameter")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wimplicit-fallthrough=0")
+if (CMAKE_COMPILER_IS_GNUCC)
+    if(NOT CMAKE_C_COMPILER_VERSION VERSION_LESS 7.0)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wimplicit-fallthrough=0")
+    endif()
+endif()
 
 list(INSERT CMAKE_MODULE_PATH 0
 	${CMAKE_CURRENT_SOURCE_DIR}/CMake

